scripts/camera/install-camera-pi.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+# This script should be run as root.
+set -eu
+
+PANUSER=${PANUSER:-huntsman}
+PANDIR=${PANDIR:-/var/huntsman}
+HOME=${HOME:-/home/${PANUSER}}
+LOGFILE="${PANDIR}/install-camera-pi.log"
+ENV_FILE="${PANDIR}/env"
+
+function command_exists() {
+  # https://gist.github.com/gubatron/1eb077a1c5fcf510e8e5
+  # this should be a very portable way of checking if something is on the path
+ # usage: "if command_exists foo; then echo it exists; fi"
+ type "$1" &>/dev/null
+}
+
+function make_directories() {
+ mkdir -p "${HOME}/.ssh"
+ mkdir -p "${PANDIR}"
+ mkdir -p "${PANDIR}/logs"
+ mkdir -p "${PANDIR}/images"
+ mkdir -p "${PANDIR}/config_files"
+ mkdir -p "${PANDIR}/.key"
+ chown -R "${PANUSER}":"${PANUSER}" "${PANDIR}"
+ chown -R "${PANUSER}":"${PANUSER}" "${HOME}"
+}
+
+function setup_env_vars() {
+ if [[ ! -f "${ENV_FILE}" ]]; then
+   echo "Writing environment variables to ${ENV_FILE}"
+   cat >>"${ENV_FILE}" <<EOF
+#### Added by install-camera-pi script ####
+export PANUSER=${PANUSER}
+export PANDIR=${PANDIR}
+export POCS=${PANDIR}/POCS
+export PANLOG=${PANDIR}/logs
+#### End install-pocs script ####
+EOF
+  fi
+  echo ". ${ENV_FILE}" >> "${HOME}/.bashrc"
+}
+
+function system_deps() {
+ apt-get update | tee -a "${LOGFILE}" 2>&1
+ apt-get --yes install \
+   wget curl \
+   git openssh-server \
+   git \
+   jq httpie \
+   byobu | tee -a "${LOGFILE}" 2>&1
+ # Add an SSH key if one doesn't exist.
+ if [[ ! -f "${HOME}/.ssh/id_rsa" ]]; then
+   echo "Adding ssh key"
+   ssh-keygen -t rsa -N "" -f "${HOME}/.ssh/id_rsa"
+ fi
+
+ # Append some statements to .bashrc
+ cat <<EOF >>/home/${PANUSER}/.bashrc
+export LANG="en_US.UTF-8"
+
+# POCS
+export PANDIR=/var/huntsman
+unsetopt share_history
+EOF
+}
+
+function enable_auto_login() {
+  # Set up autologin without password for huntsman user
+  # This is a bit of a hack but it appears to be the standard method of doing this
+  sed -i '/^ExecStart=$/d' /lib/systemd/system/getty@.service
+  sed -i "s/ExecStart=.*/ExecStart=\nExecStart=-\/sbin\/agetty -a huntsman --noclear %I \$TERM/g" /lib/systemd/system/getty@.service
+  sed -i "s/Type=idle/Type=simple/g" /lib/systemd/system/getty@.service
+
+  # We also need to disable password change on first login
+  # The easiest way of doing this is just to change the password here
+  # The password expiry status can be checked by doing $chage -l ${PAUSER}
+  usermod --password "${PANUSER}" "${PANUSER}"
```

Comment:
Is this still necessary? I saw that the expiration was removed from the `user-data` file.

Reply:
I removed it from `user-data` but it seems that ubuntu's default behaviour is to require the immediate password reset.